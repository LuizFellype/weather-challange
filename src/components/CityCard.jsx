import * as React from "react";

const CityCard = () => {
  return (
    <div />
  )
};

export default CityCard
