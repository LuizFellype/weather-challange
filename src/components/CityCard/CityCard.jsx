import * as React from "react";
import { normalizeWeatherData } from "./helpers/normalizeWeatherData";
import { WeatherInfo } from "../WeatherInfo";

const CityCard = ({ 
  name,
  temp,
  pressure,
  humidity,
  tempColorCode
}) => {
  return (
    <div className="m-2 items-center flex flex-col md:flex-row md:justify-center cursor-default">
      <div className="w-60 transition duration-500 ease-in-out transform bg-white rounded-lg hover:scale-105 cursor-pointer border flex flex-col justify-center items-center text-center p-6">
        <div className="text-md flex flex-col text-gray-900 mb-2">
          <span className="uppercase font-bold">{name}</span>
        </div>

        <div className="flex gap-1">
          <WeatherInfo label="Temp" value={temp} className={tempColorCode} />
          <WeatherInfo label="Pressure" value={pressure} />
          <WeatherInfo label="Hum" value={humidity} />
        </div>
      </div>
    </div>
  )
};

const CityCardWrapper = (props) => {
  const data = normalizeWeatherData(props)

  return <CityCard {...data} />
};

export default CityCardWrapper
