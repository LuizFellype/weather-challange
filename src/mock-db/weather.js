
export const allWeathers = [
  {
    "lat": 32.44,
    "lon": -94.04,
    "timezone": "City 1",
    "timezone_offset": -18000,
    "current": {
      "dt": 1684929490,
      "sunrise": 1684926645,
      "sunset": 1684977332,
      "temp": 5,
      "feels_like": 292.87,
      "pressure": 1014,
      "humidity": 89,
      "dew_point": 290.69,
      "uvi": 0.16,
      "clouds": 53,
      "visibility": 10000,
      "wind_speed": 3.13,
      "wind_deg": 93,
      "wind_gust": 6.71,
      "weather": [
        {
          "id": 803,
          "main": "Clouds",
          "description": "broken clouds",
          "icon": "04d"
        }
      ]
    },
  },
  {
    "lat": 33.44,
    "lon": -94.04,
    "timezone": "City 2",
    "timezone_offset": -18000,
    "current": {
      "dt": 1684929490,
      "sunrise": 1684926645,
      "sunset": 1684977332,
      "temp": 25,
      "feels_like": 292.87,
      "pressure": 1014,
      "humidity": 89,
      "dew_point": 290.69,
      "uvi": 0.16,
      "clouds": 53,
      "visibility": 10000,
      "wind_speed": 3.13,
      "wind_deg": 93,
      "wind_gust": 6.71,
      "weather": [
        {
          "id": 803,
          "main": "Clouds",
          "description": "broken clouds",
          "icon": "04d"
        }
      ]
    },
  },
  {
    "lat": 33.44,
    "lon": -94.04,
    "timezone": "City 3",
    "timezone_offset": -18000,
    "current": {
      "dt": 1684929490,
      "sunrise": 1684926645,
      "sunset": 1684977332,
      "temp": 26,
      "feels_like": 292.87,
      "pressure": 1014,
      "humidity": 89,
      "dew_point": 290.69,
      "uvi": 0.16,
      "clouds": 53,
      "visibility": 10000,
      "wind_speed": 3.13,
      "wind_deg": 93,
      "wind_gust": 6.71,
      "weather": [
        {
          "id": 803,
          "main": "Clouds",
          "description": "broken clouds",
          "icon": "04d"
        }
      ]
    },
  },
];
