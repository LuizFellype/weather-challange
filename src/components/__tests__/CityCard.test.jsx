import {render} from '@testing-library/react'
import CityCard from '../CityCard'
import {allWeathers} from '../../mock-db/weather'

const [lowTempCity, midTempCity, highTempCity] = allWeathers

describe('CityCard temperature color coding', () => {
    test('should render blue for 5째C or below temperature', () => {
        const { asFragment } = render(<CityCard data={lowTempCity}/>)
        
        expect(asFragment()).toMatchSnapshot()
    })
    test('should render orange for temperature between 5째C and 26째C', () => {
        const { asFragment } = render(<CityCard data={midTempCity}/>)
        
        expect(asFragment()).toMatchSnapshot()
    })
    test('should render red for temperature above 25째C', () => {
        const { asFragment } = render(<CityCard data={highTempCity}/>)
        
        expect(asFragment()).toMatchSnapshot()
    })
    
})