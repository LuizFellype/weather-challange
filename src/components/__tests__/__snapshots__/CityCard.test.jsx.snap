// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityCard temperature color coding should render blue for 5째C or below temperature 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`CityCard temperature color coding should render orange for temperature between 5째C and 26째C 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`CityCard temperature color coding should render red for temperature above 25째C 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
